source/libraries/ContractInterfacesGenerator.ts

```diff
@@ -0,0 +1,181 @@
+import * as fs from "async-file";
+import { Abi, AbiFunction, Primitive } from 'ethereum';
+import { ContractCompiler } from "../libraries/ContractCompiler";
+import { CompilerOutput, CompilerOutputContracts } from "solc";
+import { Configuration } from '../libraries/Configuration';
+
+
+export async function generateContractInterfaces(compiler: ContractCompiler, configuration: Configuration): Promise<String> {
```

Comment:
I would rather see this as a class (for consistency) as mentioned in the other comment.  I think I just like importing a first letter capitalized things....
```
public contract ContractInterfaceGenerator {
  constructor(configuration: Configuration, contractCompiler: ContractCompiler) {
    // TODO: assign to member variables
  }
  public function generateContractInterfaces(): Promise<string> {
    // TODO: compile contracts
    // TODO: generate string
    // TODO: write to file
    // TODO: return string
  }
  private function contractInterfacesTemplate(): string {
    // TODO
  }
}
```

Reply:
Makes sense! Will do